packages/forms/src/elements/file-list/components/File.tsx

```diff
@@ -37,16 +36,21 @@ interface IStaticFileExport<T, P>
 /**
  * @extends HTMLAttributes<HTMLDivElement>
  */
-export const File = forwardRef<HTMLDivElement, IFileProps>(({ children, type, ...props }, ref) => (
-  <StyledFile {...props} ref={ref}>
-    {type && <StyledIcon>{fileIcons[type]}</StyledIcon>}
-    {children}
-  </StyledFile>
-)) as IStaticFileExport<HTMLDivElement, IFileProps>;
+export const File = forwardRef<HTMLDivElement, IFileProps>(
+  ({ children, type, focusInset, ...props }, ref) => (
+    <StyledFile {...props} focusInset={focusInset} ref={ref}>
+      {type && <StyledFileIcon>{fileIcons[type]}</StyledFileIcon>}
+      {Children.map(children, child => (typeof child === 'string' ? <span>{child}</span> : child))}
```

Comment:
I'm not entirely enthused with this approach, but we needed a way to ensure the file name could participate as a proper flex child.